tools/validate-deployment.sh

```diff
@@ -0,0 +1,117 @@
+        echo "Resetting public key to ProctorVM "
+        az vm user update --resource-group=ProctorVMG --name=proctorVM --username azureuser --ssh-key-value $ID_RSA_PUBLIC
+    else
+        echo "[ERROR] Public key missing when resetting key for ProctorVM for $teamAAD - Subscription $subid - Exiting ..."
+    fi
+
+    scp -o StrictHostKeyChecking=no -i $ID_RSA_PRIVATE -r azureuser@$ipaddress:/home/azureuser/team_env/* ./$teamAAD/
+    if [ $? -ne 0 ]; then
+        echo "[ERROR] Getting team_env directory failed" >> $ERROR_FILE
+    fi
+    # Getting stderr and stdout from custom script extension.
+    ssh -o StrictHostKeyChecking=no -i $ID_RSA_PRIVATE azureuser@$ipaddress "bash -s" << EOF
+        sudo cp /var/lib/waagent/custom-script/download/0/stderr .;
+        sudo chown azureuser:azureuser ./stderr;
+EOF
+    scp -o StrictHostKeyChecking=no -i $ID_RSA_PRIVATE azureuser@$ipaddress:./stderr ./$teamAAD/
+    ssh -o StrictHostKeyChecking=no -i $ID_RSA_PRIVATE azureuser@$ipaddress "bash -s" << EOF
+        sudo cp /var/lib/waagent/custom-script/download/0/stdout .;
+        sudo chown azureuser:azureuser ./stdout;
+EOF
+    scp -o StrictHostKeyChecking=no -i $ID_RSA_PRIVATE azureuser@$ipaddress:./stdout ./$teamAAD/
+    # Getting deployment logs 
+    scp -o StrictHostKeyChecking=no -i $ID_RSA_PRIVATE azureuser@$ipaddress:/home/azureuser/openhack-devops-proctor/provision-team/teamdeploy.out ./$teamAAD/
+    if [ $? -ne 0 ]; then
+        echo "[ERROR] Getting teamdeploy.out file failed for subscription $subid, portal username $portalUserName, AAD $teamAAD, VM IP is $ipaddress" >> $ERROR_FILE
+    fi
+
+    grep -x "poi   \[X\]" ./$teamAAD/teamdeploy.out
+    if [ $? -ne 0 ]; then
+        echo "[ERROR] - Deployment of API poi has failed in subscription $subid, portal username $portalUserName, AAD $teamAAD - VM IP is $ipaddress" >> $ERROR_FILE 
+    fi
+
+    grep -x "user  \[X\]" ./$teamAAD/teamdeploy.out
+    if [ $? -ne 0 ]; then
+        echo "[ERROR] - Deployment of API user has failed in subscription $subid, portal username $portalUserName, AAD $teamAAD - VM IP is $ipaddress" >> $ERROR_FILE
+    fi
+
+    grep -x "trips \[X\]" ./$teamAAD/teamdeploy.out
+    if [ $? -ne 0 ]; then
+        echo "[ERROR] - Deployment of API trips has failed in subscription $subid, portal username $portalUserName, AAD $teamAAD - VM IP is $ipaddress" >> $ERROR_FILE 
+    fi
+
+    grep -x "user-java \[X\]" ./$teamAAD/teamdeploy.out
+    if [ $? -ne 0 ]; then
+        echo "[ERROR] - Deployment of API user-java has failed in subscription $subid, portal username $portalUserName, AAD $teamAAD - VM IP is $ipaddress" >> $ERROR_FILE 
+    fi
+    
+    grep -x '############ END OF TEAM PROVISION ############' ./$teamAAD/teamdeploy.out
+    if [ $? -ne 0 ]; then
+        echo "[ERROR] - Deployment of $teamAAD has failed in subscription $subid, portal username $portalUserName, AAD $teamAAD - VM IP is $ipaddress" >> $ERROR_FILE  
+    fi
+fi
+        
```

Comment:
The script should return the proper exit code if at least of the above checks failed but should run all the tests.

Reply:
@dcaro Here just adding `exit 1` should suffice right in each `if fi` condition

Reply:
This exist the script on the first fail but I'd like to have the result of all the other tests as well to know if the issue is on one API or all for example.

Reply:
@dcaro Not sure if this is what you are looking for in terms of exit code.

Reply:
The overall logic looks good, please have a look at my comment on the if statement below.